web/src/app/shared/services/classifications/classifications.service.ts

```diff
@@ -16,52 +15,50 @@ import { QueryParameters } from 'app/shared/models/query-parameters';
 @Injectable()
 export class ClassificationsService {
   private url = `${environment.taskanaRestUrl}/v1/classifications/`;
-  private classificationSaved = new Subject<number>();
-  private classificationResourcePromise: Promise<ClassificationResource>;
+  private classificationResourcePromise: Promise<ClassificationPagingList>;
   private lastDomain: string;
 
   constructor(
     private httpClient: HttpClient,
     private domainService: DomainService,
-  ) {}
+  ) {
+  }
 
-  private static classificationParameters(domain: string): QueryParameters {
+  private static classificationParameters(domain: string, type?: string): QueryParameters {
     const parameters = new QueryParameters();
     parameters.SORTBY = TaskanaQueryParameters.parameters.KEY;
     parameters.SORTDIRECTION = Direction.ASC;
     parameters.DOMAIN = domain;
+    parameters.TYPE = type;
     delete TaskanaQueryParameters.page;
     delete TaskanaQueryParameters.pageSize;
 
     return parameters;
   }
 
   // GET
-  getClassifications(classificationType?: string): Observable<Array<Classification>> {
-    return this.domainService.getSelectedDomain().pipe(
-      mergeMap(domain => this.getClassificationObservable(this.httpClient.get<ClassificationResource>(
-        `${this.url}${TaskanaQueryParameters.getQueryParameters(ClassificationsService.classificationParameters(domain))}`
-      ), classificationType)),
-      tap(() => {
-        this.domainService.domainChangedComplete();
-      })
-    );
+  getClassifications(classificationType?: string): Observable<ClassificationPagingList> {
+    return this.httpClient.get<ClassificationPagingList>(
+      `${this.url}${TaskanaQueryParameters.getQueryParameters(
+        ClassificationsService.classificationParameters(this.domainService.getSelectedDomainValue(), classificationType)
+      )}`
+    ).pipe(tap(() => this.domainService.domainChangedComplete()));
```

Comment:
As far is i can tell this call does nothing, as requestInProgress never gets set to true, so setting it to false after this call does not make sense. Maybe remove this call. Or this could be related to [TSK-1306](https://taskana.atlassian.net/browse/TSK-1306) and be removed or reworked there.

Reply:
You are right. I am breaking the requestInProgress mechanism of the domainService. The domainService and the requestInProgress mechanism should be reworked. Until that we have multiple solutions:
* revert this hunk so that we have the old mechanism in play
* remove the call

which one is preferred?

For now I'll revert this hunk.